audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractTest.java

```diff
@@ -309,22 +319,21 @@ void testFailedSuccessDeserialization() throws InterruptedException {
                 .withBootstrapServers(KafkaInitializer.kafka.getBootstrapServers());
 
         Source.fromJavaStream(() -> requests)
-                .runWith(Producer.plainSink(producerSettings), akkaMaterializer);
+                .runWith(Producer.plainSink(producerSettings), akkaMaterializer)
+                .toCompletableFuture().join();
 
         TimeUnit.SECONDS.sleep(2);
 
-        // Then check an "Success-..." file has been created
-        Set<File> expectedFileSet = Arrays.stream(new File(".").listFiles())
-                .filter(file -> file.getName().startsWith("Success"))
-                .collect(Collectors.toSet());
-        Set<File> actualFileSet = Arrays.stream(new File(auditServiceConfigProperties.getErrorDirectory()).listFiles())
+        // Then check a "Success-..." file has been created
+        after = Arrays.stream(new File(auditServiceConfigProperties.getErrorDirectory()).listFiles())
                 .filter(file -> file.getName().startsWith("Success"))
-                .collect(Collectors.toSet());
+                .collect(Collectors.toSet()).size();
+
+        assertThat(after).as("Check at least 1 'Success' file has been created").isEqualTo(successBefore + 1);
+    }
 
-        assertAll(
-                () -> assertThat(actualFileSet).as("Check at least 1 'Success' file has been created").isNotEmpty(),
-                () -> assertThat(actualFileSet).as("Check the 'Success' files are the same").containsAll(expectedFileSet)
-        );
+    private void sleep(final int sleep) throws InterruptedException {
```

Comment:
I was more expecting:
```
private void waitForServiceProcessing() throws InterruptedException {
    TimeUnit.SECONDS.sleep(1);
}
```
so if we want to make that time shorter/longer, it's only in that one spot

Reply:
Updated this